pkg/util/controller_finder.go

```diff
@@ -228,10 +227,9 @@ func (r *ControllerFinder) getStatefulSetLikeWorkload(namespace string, ref *rol
 		return nil, nil
 	}
 
-	unifiedObject := &unstructured.Unstructured{}
-	unifiedObjectKey := types.NamespacedName{Name: ref.Name, Namespace: namespace}
-	unifiedObject.SetGroupVersionKind(schema.FromAPIVersionAndKind(ref.APIVersion, ref.Kind))
-	err := r.Get(context.TODO(), unifiedObjectKey, unifiedObject)
+	key := types.NamespacedName{Name: ref.Name, Namespace: namespace}
+	set := GetEmptyWorkloadObject(schema.FromAPIVersionAndKind(ref.APIVersion, ref.Kind))
+	err := r.Get(context.TODO(), key, set)
```

Comment:
set can be nil if the workload is not supported, shall we deal with the nil case